src/components/App.tsx

```diff
@@ -44,115 +32,160 @@ export default function App() {
   // check localStorage if it contains a - from before - chosen network, if not
   // retrieve the 1st available network from the given ones, else (rare and wrong case)
   // default to polkadot
-  const [chain, setChain] = useState<string>(
-      location ||
-      localNetwork && localNetwork.toLowerCase() ||
-      Object.keys(allChains)[0] ||
-      "polkadot"
+  const currentName =
+    (Object.keys(allChains).includes(location) && location) ||
+    localNetwork ||
+    Object.keys(allChains)[0] ||
+    "polkadot";
+
+  const [currentNetwork, setCurrentNetwork] = useState<string>(currentName);
+  const [metadataQr, setMetadataQr] = useState<QrInfo>(
+    allChains[currentName]?.metadataQr
```

Comment:
I mean, could we instead of 
```
const [metadataQr, setMetadataQr] = useState<QrInfo>(
    allChains[currentName]?.metadataQr
  );
```
do smth like 
```
const qr = allChains[currentName]?.metadataQr;
```

I don't think there is a need for `setMetadataQr`